src/app/Http/Controllers/BusinessTripController.php

```diff
@@ -77,4 +85,203 @@ public function confirm() {
     public function close() {
 
     }
+
+    public function exportPdf(int $tripId, DocumentType $documentType): JsonResponse | BinaryFileResponse
+    {
+        $trip = BusinessTrip::find($tripId);
+        if (!$trip && $tripId !== 0) {
+            return response()->json(['error' => 'Business trip not found'], 404);
+        }
+
+        $templateName = $documentType->value;
+        $templatePath = Storage::disk('pdf-templates')
+            ->path($templateName);
+
+        if (Storage::disk('pdf-templates')->missing($templateName)) {
+            Log::error("PDF template file does not exist at path: " . $templatePath);
+            return response()->json(['error' => 'PDF template file not found'], 404);
+        }
+
+        $outputPath = Storage::disk('pdf-exports')
+            ->path(uniqid('', true) . '.pdf');
+
+        $data = [];
+        switch ($documentType) {
+            case 'cestne_vyhlasenie_k_zahranicnej_pc.pdf':
+                $data = [
+                    'order_number' => $trip->sofia_id,
+                    'trip_duration' => '?', //$trip->'?', // tu potrebujem pomôcť, ktorý stĺpec z databázy to je
+                    'adress' => $trip->place,
+                    'name' => $trip->user->first_name . ' ' . $trip->user->last_name,
+                ];
+                break;
+
+            case DocumentType::COMPENSATION_AGREEMENT:
+                $data = [
+                    'first_name' => $trip->user->first_name,
+                    'last_name' => $trip->user->last_name,
+                    'academic_degree' => $trip->user->academic_degrees,
+                    'address' => $trip->user->address,
+                    'contribution1' => '?', // v databáze neexistuje prepojenie na tabuľku s prínosmi pre fakultu
+                    'contribution2' => '?', // -||-
+                    'contribution3' => '?', // -||-
+                    'department' => $trip->user->department,
+                    'place' => $trip->country->name . ', ' . $trip->place,
+                    'datetime_start' => $trip->datetime_start->format('Y-m-d'),
+                    'datetime_end' => $trip->datetime_end->format('Y-m-d'),
+                    'transport' => $trip->transport->name,
+                    'trip_purpose' => $trip->tripPurpose->name,
+                    'fund' => $trip->sppSymbol->fund,
+                    'functional_region' => $trip->sppSymbol->functional_region,
+                    'financial_centre' => $trip->sppSymbol->financial_centre,
+                    'spp_symbol' => $trip->sppSymbol->spp_symbol,
+                    'account' => $trip->sppSymbol->account,
+                    'grantee' => $trip->sppSymbol->grantee,
+                    'iban' => $trip->iban,
+                    'bank_name' => '?',
```

Comment:
toto nemusime riesit snad, nemame na to ani stlpec, takze zatial urcite netreba vyplnat